<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="Cucumber" time="3.619" tests="4" skipped="0" failures="3" errors="0" timestamp="2025-09-29T11:00:55.851150600Z">
<testcase classname="API_Web_React_E-E_Validation" name="Post_Request" time="2.759">
<failure type="java.lang.AssertionError" message="1 expectation failed.
Expected status code &lt;201&gt; but was &lt;500&gt;.
">
<![CDATA[java.lang.AssertionError: 1 expectation failed.
Expected status code <201> but was <500>.

	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:57)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$1.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
	at io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)
	at io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:135)
	at io.restassured.specification.ResponseSpecification$statusCode$0.callCurrent(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:143)
	at io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:89)
	at stepdefinitions.React_API_P_G_P_D.i_send_a_post_request_to_the_api_with_endpoint_and_name(React_API_P_G_P_D.java:46)
	at ✽.I send a POST request to the API with endpoint "/api/trainings" and name 7(file:///C:/Users/aarulfrancis/Downloads/FirstProject/demo/src/test/resources/features/React_API.feature:7)
]]>
</failure>
<system-out><![CDATA[
Given The base URI of API is "http://10.192.190.130:5000"...................passed
When I send a POST request to the API with endpoint "/api/trainings" and name 7.failed
Then I should receive the post response.....................................skipped
And verify the status code is "201" and post request is successful in API...skipped
And validate if a new training is CREATED in the Web........................skipped
]]></system-out>
</testcase>
<testcase classname="API_Web_React_E-E_Validation" name="Put_Request" time="0.094">
<failure type="java.lang.AssertionError" message="1 expectation failed.
Expected status code &lt;200&gt; but was &lt;404&gt;.
">
<![CDATA[java.lang.AssertionError: 1 expectation failed.
Expected status code <200> but was <404>.

	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:512)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$1.call(Unknown Source)
	at io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:696)
	at io.restassured.internal.ResponseSpecificationImpl.this$2$validateResponseIfRequired(ResponseSpecificationImpl.groovy)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:198)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:135)
	at io.restassured.specification.ResponseSpecification$statusCode$0.callCurrent(Unknown Source)
	at io.restassured.internal.ResponseSpecificationImpl.statusCode(ResponseSpecificationImpl.groovy:143)
	at io.restassured.internal.ValidatableResponseOptionsImpl.statusCode(ValidatableResponseOptionsImpl.java:89)
	at stepdefinitions.React_API_P_G_P_D.i_send_a_put_request_to_the_api_with_endpoint_and_name(React_API_P_G_P_D.java:106)
	at ✽.I send a PUT request to the API with endpoint "/api/trainings/" and name 6(file:///C:/Users/aarulfrancis/Downloads/FirstProject/demo/src/test/resources/features/React_API.feature:13)
]]>
</failure>
<system-out><![CDATA[
Given The base URI of API is "http://10.192.190.130:5000"...................passed
When I send a PUT request to the API with endpoint "/api/trainings/" and name 6.failed
Then I should receive the put response......................................skipped
And the status code contains "200" and put request is successful in API.....skipped
And validate if the corresponding record got UPDATED in the Web.............skipped
]]></system-out>
</testcase>
<testcase classname="API_Web_React_E-E_Validation" name="Get_Request" time="0.421">
<system-out><![CDATA[
Given The base URI of API is "http://10.192.190.130:5000"...................passed
When I send a GET request to the API with endpoint "/api/trainings".........passed
Then I should receive the response..........................................passed
And the status code is "200"................................................passed
]]></system-out>
</testcase>
<testcase classname="API_Web_React_E-E_Validation" name="Delete_Request" time="0.052">
<failure type="java.lang.NullPointerException" message="Cannot invoke &quot;org.openqa.selenium.WebDriver.get(String)&quot; because &quot;stepdefinitions.React_API_P_G_P_D.driver&quot; is null">
<![CDATA[java.lang.NullPointerException: Cannot invoke "org.openqa.selenium.WebDriver.get(String)" because "stepdefinitions.React_API_P_G_P_D.driver" is null
	at stepdefinitions.React_API_P_G_P_D.validate_if_the_corresponding_record_got_deleted_in_the_web(React_API_P_G_P_D.java:191)
	at ✽.validate if the corresponding record got DELETED in the Web(file:///C:/Users/aarulfrancis/Downloads/FirstProject/demo/src/test/resources/features/React_API.feature:27)
]]>
</failure>
<system-out><![CDATA[
Given The base URI of API is "http://10.192.190.130:5000"...................passed
When I send a DELETE request to the API with endpoint "/api/trainings/".....passed
Then I should receive the delete response...................................passed
And verify if the status code has "204" and delete request is successful in API.passed
And validate if the corresponding record got DELETED in the Web.............failed
]]></system-out>
</testcase>
</testsuite>
